zerver/tests/test_messages.py

```diff
@@ -1395,6 +1395,22 @@ def test_personal_message(self) -> None:
                                                      "to": self.example_email("othello")})
         self.assert_json_success(result)
 
+    def test_personal_message_by_id(self) -> None:
+        """
+        Sending a personal message to a valid user ID is successful.
+        """
+        self.login(self.example_email("hamlet"))
+        result = self.client_post(
+            "/json/messages",
+            {
+                "type": "private",
+                "content": "Test message",
+                "client": "test suite",
+                "to": ujson.dumps([self.example_user("othello").id])
+            }
+        )
+        self.assert_json_success(result)
```

Comment:
Let's include a `get_last_message` check that it actually sent a PM to Othello.

Reply:
Done! :)

Reply:
This should also have a test for group PMs; I'll just add that while merging.